src/main/java/org/passay/dictionary/FileWordList.java

```diff
@@ -87,98 +73,31 @@ public FileWordList(final RandomAccessFile raf, final boolean caseSensitive)
   public FileWordList(final RandomAccessFile raf, final boolean caseSensitive, final int cachePercent)
     throws IOException
   {
-    if (cachePercent < 0 || cachePercent > HUNDRED_PERCENT) {
-      throw new IllegalArgumentException("cachePercent must be between 0 and 100 inclusive");
-    }
-    file = raf;
-    if (caseSensitive) {
-      comparator = WordLists.CASE_SENSITIVE_COMPARATOR;
-    } else {
-      comparator = WordLists.CASE_INSENSITIVE_COMPARATOR;
-    }
-    synchronized (file) {
-      file.seek(0L);
-
-      String a;
-      String b = null;
-      while ((a = file.readLine()) != null) {
-        if (b != null && comparator.compare(a, b) < 0) {
-          throw new IllegalArgumentException("File is not sorted correctly for this comparator");
-        }
-        b = a;
-        size++;
+    super(raf, caseSensitive, cachePercent);
+    long pos = 0;
+    file.seek(pos);
+
+    final long fileBytes = file.length();
+    final long cacheSize = (fileBytes / HUNDRED_PERCENT) * cachePercent;
+    final long cacheOffset = cacheSize == 0 ? fileBytes : cacheSize > fileBytes ? 1 : fileBytes / cacheSize;
```

Comment:
It took a fair bit of study to understand the purpose of this variable, and I believe the name added to the difficulty. Consider renaming `cacheModulus` or `cacheFrequency`.